src/content-helper/editor-sidebar/title-suggestions/provider.ts

```diff
@@ -1,75 +1,75 @@
 /**
  * WordPress dependencies
  */
-import apiFetch from '@wordpress/api-fetch';
 import { addQueryArgs } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
-import { PersonaProp, getPersonaLabel } from '../../common/components/persona-selector';
-import { ToneProp, getToneLabel } from '../../common/components/tone-selector';
-import {
-	ContentHelperError,
-	ContentHelperErrorCode,
-} from '../../common/content-helper-error';
-
-/**
- * Specifies the form of the response returned by the
- * `content-suggestions/suggest-headline` WordPress REST API endpoint.
- *
- * @since 3.12.0
- */
-interface SuggestHeadlineApiResponse {
-	error?: Error;
-	data: string[],
-}
+import { BaseProvider } from '../../common/base-provider';
+import { getToneLabel, ToneProp } from '../../common/components/tone-selector';
+import { getPersonaLabel, PersonaProp } from '../../common/components/persona-selector';
 
 /**
  * Returns data from the `content-suggestions/suggest-headline` WordPress REST API
  * endpoint.
  *
  * @since 3.12.0
  */
-export class TitleSuggestionsProvider {
+export class TitleSuggestionsProvider extends BaseProvider {
+	/**
+	 * The singleton instance of the TitleSuggestionsProvider.
+	 *
+	 * @since 3.15.0
+	 */
+	private static instance: TitleSuggestionsProvider;
+
+	/**
+	 * Returns the singleton instance of the TitleSuggestionsProvider.
+	 *
+	 * @since 3.15.0
+	 *
+	 * @return {TitleSuggestionsProvider} The singleton instance.
+	 */
+	public static getInstance(): TitleSuggestionsProvider {
+		if ( ! this.instance ) {
+			this.instance = new TitleSuggestionsProvider();
+		}
+
+		return this.instance;
+	}
+
 	/**
 	 * Returns a list of suggested titles for the given content.
 	 *
-	 * @param {string }     content The content to generate titles for.
-	 * @param {number}      limit   The number of titles to return. Defaults to 3.
-	 * @param {ToneProp}    tone    The tone to use when generating the titles.
-	 * @param {PersonaProp} persona The persona to use when generating the titles.
+	 * @param { string }      content The content to generate titles for.
+	 * @param { number }      limit   The number of titles to return. Defaults to 3.
+	 * @param { ToneProp }    tone    The tone to use when generating the titles.
+	 * @param { PersonaProp } persona The persona to use when generating the titles.
 	 *
+	                                  <<<<<<< HEAD
+	                                  <<<<<<< HEAD
+	 * @return { Promise<string[]> } The resulting list of titles.
+	  =======
 	 * @return {Promise<string[]>} The resulting list of titles.
+	  >>>>>>> 3696940e (TypeScript DocBlocks: Remove spaces in braces)
+	  =======
+	 * @return {Promise<string[]>} The resulting list of titles.
+	  >>>>>>> develop
```

Comment:
Some funky stuff happening here 🙂

Reply:
It also seems some additional things were reverted in https://github.com/Parsely/wp-parsely/pull/2401/commits/9396e93e48a58cb7def1b091d9638ddb15faf1bd? A bit strange.

Reply:
The revert seems to be mostly the changes of https://github.com/Parsely/wp-parsely/pull/2401/commits/e9d8c5f81ad6fa1af658792d4c305bbfc28241d2.

Reply:
This is weird! Let me clean this up.